ntoskrnl/kd/kdio.c

```diff
@@ -574,19 +574,19 @@ KdpPrintString(
     PLIST_ENTRY CurrentEntry;
     PKD_DISPATCH_TABLE CurrentTable;
     PCHAR String;
+    CHAR StringBuffer[512];
 
     if (!KdpDebugMode.Value) return 0;
 
-    Length = min(Length, 512);
+    Length = min(Length, sizeof(StringBuffer));
 
     if (ExGetPreviousMode() != KernelMode)
     {
         _SEH2_TRY
         {
             ProbeForRead(UnsafeString, Length, 1);
-            String = _alloca(Length + 1);
+            String = StringBuffer;
             RtlCopyMemory(String, UnsafeString, Length);
-            String[Length] = ANSI_NULL;
```

Comment:
It would still be nice to have a null terminated string?

Reply:
Everything so far concerns counted strings, so it seems more consistent to remain with that.

Reply:
I introduced this with the last round of changes because I was too lazy to find and fix all the functions that relied on this. Now that those are fixed, I don't see a reason why it needs to be null-terminated. That just encourages bugs in the follow-up functions.